src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SystemWebAdaptersExtensions.cs

```diff
@@ -17,6 +16,7 @@ public static class SystemWebAdaptersExtensions
     public static ISystemWebAdapterBuilder AddSystemWebAdapters(this IServiceCollection services)
     {
         services.AddHttpContextAccessor();
+        services.AddSingleton<IHttpRuntime>(_ => HttpRuntimeFactory.Create());
```

Comment:
```suggestion
        services.AddSingleton<IHttpRuntime>(HttpRuntimeFactory.Create);
```

Reply:
That wouldn't work - there's no overload that takes a `Func<T>` but rather it expects a `Func<IServiceProvider,T>`

Reply:
Ah, good point. I missed that it expects a service provider.